claim/claim.c

```diff
@@ -110,27 +110,18 @@ void load_claiming_state(void)
     }
 
     char filename[FILENAME_MAX + 1];
-    struct stat statbuf;
-
     snprintfz(filename, FILENAME_MAX, "%s/claim.d/claimed_id", netdata_configured_user_config_dir);
 
-    // check if the file exists
-    if (lstat(filename, &statbuf) != 0) {
-        info("lstat on File '%s' failed reason=\"%s\". Setting state to AGENT_UNCLAIMED.", filename, strerror(errno));
-        return;
-    }
-
-    FILE *f = fopen(filename, "rt");
-    if (unlikely(f == NULL)) {
-        error("File '%s' cannot be opened. Setting state to AGENT_UNCLAIMED.", filename);
+    long bytes_read;
+    claimed_id = read_by_filename(filename, &bytes_read);
+    if (!claimed_id) {
+        info("Unable to load '%s', setting state to AGENT_UNCLAIMED", filename);
         return;
     }
 
-    claimed_id = callocz(1, statbuf.st_size + 1);
-    size_t bytes_read = fread(claimed_id, 1, statbuf.st_size, f);
-    claimed_id[bytes_read] = 0;
-    fclose(f);
     if (bytes_read == 0) {
+        freez(claimed_id);
+        claimed_id = NULL;
         info("File '%s' has no contents. Setting state to AGENT_UNCLAIMED.", filename);
         return;
     }
```

Comment:
`bytes_read` can no longer be 0. `read_by_filename()` has already returned `NULL` before that.

Moreover, it is impossible for `read_by_filename()` to return a valid `claimed_id` pointer while at the same type setting `&bytes_read` to 0.

Finally, `bytes_read` is not initialized and `read_by_filename()` is not guaranteed to always set the pointer.

I think we can remove the whole `if(){` statement.